### PR TITLE
Add support for virtual properties with a private accessor in an interface

### PR DESCRIPTION
**Pull request by AlekseyTs**
Relevant LDM notes https://github.com/dotnet/csharplang/blob/master/meetings/2018/LDM-2018-10-17.md

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review.

File: src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs [hunk 1]
```diff
@@ -27033,6 +27680,62 @@ void Validate(ModuleSymbol m)
...
+            return 1;
+        }
+    }
+    int I4.M1 
+    {
+        set => System.Console.WriteLine(""I4.M1"");
+    }
+}
+
+public interface I3 : I1
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs]: >I3 [](start = 17, length = 2)

nit: `I3` seems unused

File: src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs [hunk 2]
```diff
@@ -13295,81 +13296,416 @@ public int P5
...
+        I4 i4 = new Test44();
+        I5 i5 = new Test55();
+        I6 i6 = new Test66();
+        i3.M2();
+        i4.M2();
+        i5.M2();
+        i6.M2();
+    }
+}
+interface Test3 : I3
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs]: >Test3 [](start = 10, length = 5)

Perhaps `ITest3`?

File: src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs (hunk 1 above)
Comment (AlekseyTs) [reply src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs]: > nit: I3 seems unused

Its presence is a use. It doesn't target any specific problem, but doesn't hurt, I believe.

---
In reply to: [257399182](https://github.com/dotnet/roslyn/pull/33413#discussion_r257399182) [](ancestors = 257399182)

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review, need a second sign-off.

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review, need a second sign-off.

**Comment by AlekseyTs** [comment]
@dotnet-bot Test this please

**Pull request merged**